### PR TITLE
Add ADR table to schema

### LINKED ISSUE #3799: Add adverse drug reactions to schema + sync
## Is your feature request related to a problem? Please describe.

There is a table being added in mobile: https://github.com/sussol/msupply/issues/8244

## Describe the solution you'd like

- Add this table to the mobile schema
- Add this to outgoing (and incoming sync??)

## Implementation

As usual

## Describe alternatives you've considered

None!

## Additional context

None!

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3799 

## Change summary

- Adds ADR table with sync

## Testing
*Ensure this pr is merged: https://github.com/openmsupply/mobile/issues/3795*

- [ ] ADR table syncs TO desktop

### Related areas to think about

N/A

### CODE REVIEW

**Comment by mark-prins** [comment]
Maybe ensure that this PR https://github.com/sussol/msupply/pull/8291 is merged too?

File: src/sync/outgoingSyncUtils.js [hunk 1]
```diff
@@ -417,6 +417,18 @@ const generateSyncData = (settings, recordType, record) => {
...
+    case 'AdverseDrugReaction': {
+      return {
+        ID: record.id,
+        form_schema_ID: record.formSchema?.id ?? '',
+        name_ID: record.name?.id ?? '',
+        user_ID: record.user?.id ?? '',
+        store_ID: settings.get(THIS_STORE_ID),
+        entry_date: getDateString(record.entryDate),
+        entry_time: getTimeString(record.entryDate),
+        data: record._data,
```
Comment (mark-prins) [inline src/sync/outgoingSyncUtils.js:429]: shouldn't this be `data`?

Comment (joshxg) [reply src/sync/outgoingSyncUtils.js:429]: 🤔  `_data` is the string so I think we send it like this to be parsed on the other side

Comment (mark-prins) [reply src/sync/outgoingSyncUtils.js:429]: Should we change `NameNote` then? that one is sending the object

**Review by mark-prins — APPROVED** [review]
Looking good!

File: src/sync/outgoingSyncUtils.js (hunk 1 above)
Comment (joshxg) [reply src/sync/outgoingSyncUtils.js:429]: ah.. hmm.. I do remember testing this against the msupply branch and receiving the data, but I will make sure to give this stringifying a proper test again

**Pull request merged**